hardware-wallet-tests/cardano-app-already-connected.ts

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable jest/no-standalone-expect */
+import expect from 'expect';
+
+import {
+  createAndRegisterHardwareWalletChannels,
+  createHardwareWalletConnectionChannel,
+  initLedgerChannel,
+  createSequentialPromptMessages,
+  createCardanoAppChannel,
+  createGetPublicKeyChannel,
+  ipcRenderer,
+} from './utils';
+
+export const run = () => {
+  expect.assertions(3);
+
+  createSequentialPromptMessages([
+    'Plug Ledger Nano S to your computer',
+    'Start Cardano APP on Nano S',
+    'Run the test again with Cardano App opened',
+    'Export the public key',
+  ]);
+
+  createAndRegisterHardwareWalletChannels();
+
+  const cardanoAppChannel = createCardanoAppChannel();
+  const publicKeyChannel = createGetPublicKeyChannel();
+  const hardwareWalletConnectionChannel = createHardwareWalletConnectionChannel();
+
+  return new Promise((resolve) => {
+    hardwareWalletConnectionChannel.onReceive(
+      async (params: { path: string }) => {
+        expect(params).toEqual({
+          disconnected: expect.any(Boolean),
+          deviceType: expect.any(String),
+          deviceId: null,
+          deviceModel: expect.any(String),
+          deviceName: expect.any(String),
+          path: expect.any(String),
+        });
```

Comment:
This won't work because at this phase the test script is setting up the channels event listeners so when `initLedgerChannel` channel is invoked the message can be consumed.

Reply:
I am not sure I understood the problem correctly.. Can't you do the following?
```
  const params = await new Promise<{ path: string }>((resolve) => {
    hardwareWalletConnectionChannel.onReceive(resolve);
    initLedgerChannel();
  });
```
I think it would work the same.